services/navigation/builtin/builtin.go

```diff
@@ -189,67 +219,123 @@ func (svc *builtIn) Reconfigure(ctx context.Context, deps resource.Dependencies,
 		return err
 	}
 
+	// Set optional variables
+	metersPerSec := defaultLinearVelocityMPerSec
+	if metersPerSec != 0 {
+		metersPerSec = svcConfig.MetersPerSec
+	}
+	degPerSec := defaultAngularVelocityDegsPerSec
+	if svcConfig.DegPerSec != 0 {
+		degPerSec = svcConfig.DegPerSec
+	}
+	positionPollingFrequencyHz := defaultPositionPollingFrequencyHz
+	if svcConfig.PositionPollingFrequencyHz != 0 {
+		positionPollingFrequencyHz = svcConfig.PositionPollingFrequencyHz
+	}
+	obstaclePollingFrequencyHz := defaultObstaclePollingFrequencyHz
+	if svcConfig.ObstaclePollingFrequencyHz != 0 {
+		obstaclePollingFrequencyHz = svcConfig.PositionPollingFrequencyHz
+	}
+	planDeviationM := defaultPlanDeviationM
+	if svcConfig.PlanDeviationM != 0 {
+		planDeviationM = svcConfig.PlanDeviationM
+	}
+	replanCostFactor := defaultReplanCostFactor
+	if svcConfig.ReplanCostFactor != 0 {
+		replanCostFactor = svcConfig.ReplanCostFactor
+	}
+
+	motionServiceName := resource.DefaultServiceName
+	if svcConfig.MotionServiceName != "" {
+		motionServiceName = svcConfig.MotionServiceName
+	}
+	mapType := defaultMapType
+	if svcConfig.MapType != "" {
+		mapType, err = navigation.StringToMapType(svcConfig.MapType)
+		if err != nil {
+			return err
+		}
+	}
+	storeCfg := navigation.StoreConfig{Type: defaultStoreType}
```

Comment:
@biotinker can you confirm this method of defining a default store is valid (From what I gather from the `store.go` code, it doesn't need a `StoreConfig{Config:XXX}` when using memory (which is what the default storeType is)

Reply:
That matches my understanding, yes.